codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/BuilderGenerator.kt

```diff
@@ -136,12 +136,15 @@ class BuilderGenerator(
         member: MemberShape,
         memberName: String
     ) {
-        // Render a `set_foo` method. This is useful as a target for code generation, because the argument type
-        // is the same as the resulting member type, and is always optional.
-        val inputType = outerType.asOptional()
+        // Render a `set_foo` method.
+        val (inputType, inputVal) = if (symbolProvider.isRequiredTraitHandled(member)) {
+            outerType to "Some(input)"
+        } else {
+            outerType.asOptional() to "input"
+        }
```

Comment:
same here, can use the output type + `isOptional`

Reply:
If you need to bridge a possibly optional input with a possible optional output I'd do something like:

fn bridgeOptionality(input: Symbol, output: Symbol, field: String) {
  if (input.isOptional() && output.isOptional()) { ... } // the rest of the 4 cases etc.
}

Reply:
Same as above, I believe your suggestion makes sense. Will fix.